code.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// 全局变量命名不规范
+var global_var int
+var GlobalVar2 string
+
+// 常量命名错误
+const max_SIZE = 100
+const minsize = 10
+
+// 1. 内存泄漏 - goroutine泄漏
+func memoryLeak() {
+	ch := make(chan int)
+	go func() {
+		for {
+			select {
+			case <-ch:
+				return
+			default:
+				// 这个goroutine永远不会退出
+				time.Sleep(time.Second)
+			}
+		}
+	}()
+	// 忘记关闭channel
+}
+
+// 2. 竞态条件
+var counter int
+
+func raceCondition() {
+	for i := 0; i < 1000; i++ {
+		go func() {
+			counter++ // 没有锁保护
```

Comment:
Protect counter variable with mutex to prevent race condition. Add 'var mu sync.Mutex' at package level.

```suggestion
		mu.Lock()
		counter++
		mu.Unlock()
```